File: Dec1/dec1.go
package dec1

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {


	fmt.Println(partone("input.txt"))
	fmt.Println(parttwo("input.txt"))
}

func partone(fileName string) (answer int) {

	file, _ := os.Open(fileName)

	scanner := bufio.NewScanner(file)
	var left []int
	var right []int

	for scanner.Scan(){
		line := scanner.Text()

		vals := strings.Split(line, "   ")

		val1, _ := strconv.Atoi(vals[0])
		val2, _ := strconv.Atoi(vals[1])
		
		left = append(left, val1)
		right = append(right, val2)
	}

	sort.Sort(sort.IntSlice(left))
	sort.Sort(sort.IntSlice(right))

	for i, _ := range left {

		if i < len(right){
			if right[i] > left[i] {
				answer += int(right[i]) - int(left[i])
			} else {
				answer += int(left[i]) - int(right[i])
			}
		}
	}

	return answer
}

func parttwo(fileName string) (answer int) {

	file, _ := os.Open(fileName)

	scanner := bufio.NewScanner(file)
	var left []int
	var right []int

	for scanner.Scan(){
		line := scanner.Text()

		vals := strings.Split(line, "   ")
		val1, _ := strconv.Atoi(vals[0])
		val2, _ := strconv.Atoi(vals[1])
		
		left = append(left, val1)
		right = append(right, val2)
	}

	sort.Sort(sort.IntSlice(left))
	sort.Sort(sort.IntSlice(right))

	for _, v := range left {

		count := 0

		for j := 0; j < len(right); j ++ {
			if right[j] == v {
				count ++
			}
		}

		answer += count*v
	}

	return answer
}

File: Dec1/dec1.py
import os
file_path = os.path.dirname(os.path.realpath(__file__)) + "/input.txt"
file_input = open(file_path, "r").read().strip()
file_lines = file_input.split("\n")
answer = 0

left_part = []
right_part = []

for line in file_lines:
  
  parts = line.split("   ")

  left_part.append(parts[0])
  right_part.append(parts[1])


left_part = sorted(left_part)
right_part = sorted(right_part)


for i in range(len(left_part)):
  if i < len(right_part):
    if right_part[i] > left_part[i]:
      answer += int(right_part[i]) - int(left_part[i])
    else :
      answer += int(left_part[i]) - int(right_part[i])


print(answer)